wallet/src/main/java/com/android/identity_credential/wallet/ui/destination/addtowallet/AddToWalletScreen.kt

```diff
@@ -100,17 +127,52 @@ fun AddToWalletScreen(
                 Logger.e(TAG, "Error loading issuers", loadingIssuerDisplayError.value!!)
                 AddToWalletScreenLoadingError(loadingIssuerDisplayError.value!!)
             } else {
-                AddToWalletScreenWithIssuerDisplayDatas(
-                    provisioningViewModel,
-                    onNavigate,
-                    documentStore,
-                    walletServerProvider,
-                    settingsModel,
-                    issuerDisplayDatas,
-                )
+                // compose ScanQrDialog when user taps on "Scan Credential Offer"
+                if (showQrScannerDialog.value) {
+                    ScanQrDialog(
+                        modifier = Modifier.fillMaxWidth(0.9f).fillMaxHeight(0.6f),
+                        title = stringResource(R.string.credential_offer_scan),
+                        description = stringResource(id = R.string.credential_offer_details),
+                        onScannedQrCode = { qrCodeText ->
+                            // scanned text is expected to be an encoded Url
+                            val decodedUrl = URLDecoder.decode(qrCodeText, "UTF-8")
+                            val urlQuery = decodedUrl.split("?")[1]
+                            // extract Credential Issuer Uri (issuing authority path) and credential id (pid-mso-mdoc, pid-sd-jwt)
+                            extractCredentialIssuerData(urlQuery).let { (credentialIssuerUri, credentialConfigurationId) ->
+                                // initiate getting issuing authority dynamically from specified Issuer Uri and Credential Id
+                                initiateCredentialOfferIssuance(
```

Comment:
How do we know that the QR code we scanned is a credential offer?
What if it's the URL of the menu for the restaurant I'm eating at?

Reply:
Ah! Good catch! The logic used in `handleOid4vciCredentialOfferIntent` from `MainActivity` 
```
 if (intent.dataString?.startsWith(WalletApplicationConfiguration.OID4VCI_CREDENTIAL_OFFER_URL_SCHEME) == true) {
```
was lost in the process of moving away from a Composable screen and into a global function. I've added this fix.